python3Packages.tox: 3.27.1 -> 4.0.16

luizirber opened the pull request:
###### Description of changes

(Keeping this as a draft PR while waiting for feedback, because there are many changes involved here...)

Bump `tox` to new major version, bringing along updates for dependencies.

Main question: should I break into multiple PRs instead (especially for `packaging` and `virtualenv`, since they trigger many updates)? Or is using one commit per package updated enough?

Tox 4 changelog and breaking changes:
https://tox.wiki/en/latest/changelog.html#v4-0-16-2022-12-20
https://tox.wiki/en/latest/faq.html#breaking-changes-in-tox-4

Load bearing packages that trigger many updates:
https://packaging.pypa.io/en/latest/changelog.html#id1
https://virtualenv.pypa.io/en/latest/changelog.html#v20-17-1-2022-12-05

Updated packages:
https://github.com/tox-dev/py-filelock/releases/tag/3.8.2
https://github.com/chardet/chardet/releases/tag/5.1.0

New package:
https://github.com/tox-dev/pyproject-api/blob/main/docs/changelog.rst#v121---2022-12-04

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

dotlambda committed:
python3Packages.pythonImportsCheck: set $PYTHONPATH

Don't rely on the installPhase doing so.

dotlambda committed:
python310Packages.pytest-xdist: 2.5.0 -> 3.1.0

https://github.com/pytest-dev/pytest-xdist/blob/v3.1.0/CHANGELOG.rst

bbjubjub2494 committed:
python3Packages.typing-extensions: 4.3.0 -> 4.4.0

dotlambda committed:
python310Packages.pytest-forked: add setup hook

mweinelt committed:
python39Packages: Stop recursing into package set

For NixOS 23.05 we are shifting support away from 3.9 and towards 3.10
and 3.11.

mweinelt committed:
python311Packages: Recurse into package set

Python 3.11 is the latest stable release that we are going to support,
together with 3.10, in NixOS 23.05. The default python version will
remain at 3.10, but 3.11 packages will be built on Hydra and provided
from the binary cache.

mweinelt committed:
python311Packages.tensorflow-bin: Disable

The upstream does not support 3.11 yet and as such has not provided any
compatible wheels.

mweinelt committed:
python3Packages.setuptools: 65.3.0 -> 65.6.3

https://setuptools.pypa.io/en/stable/history.html#v65-6-3

mweinelt committed:
python3Packages.pip: 22.2.2 -> 22.3.1

https://pip.pypa.io/en/stable/news/#v22-3-1

mweinelt committed:
python3Packages.hypothesis: 6.54.5 -> 6.61.0

https://hypothesis.readthedocs.io/en/latest/changes.html#v6-61-0

mweinelt committed:
python3Packages.flit-scm: Trim dependencies and refactor

mweinelt committed:
python3Packages.exceptiongroup: Disable tests on python310

Since pytest now requires exceptiongroup we can't use it to test the
package, because it causes an infinite recursion.

mweinelt committed:
python3Packages.pytest: 7.1.3 -> 7.2.0

https://github.com/pytest-dev/pytest/releases/tag/7.2.0

dotlambda reviewed (changes requested):
Please split into one commit per package with correctly formatted committ messages, e.g. `python310Packages.tox: 3.27.1 -> 4.0.16`.

mweinelt committed:
python3Packages.pytest-mock: 3.8.2 -> 3.10.0

https://github.com/pytest-dev/pytest-mock/releases/tag/v3.10.0
https://github.com/pytest-dev/pytest-mock/releases/tag/v3.9.0

mweinelt committed:
python3Packages.cffi: Apply patches for pytest 7.2.0 compat

mweinelt committed:
python3Packages.pytest-xprocess: 0.20.0 -> 0.21.0

https://github.com/pytest-dev/pytest-xprocess/blob/0.21.0/CHANGELOG.rst

mweinelt committed:
python3Packages.execnet: Fix compat with pytest 7.2.0

mweinelt committed:
python3Packages.flit: 3.7.1 -> 3.8.0

https://flit.pypa.io/en/stable/history.html#version-3-8

luizirber committed:
python3Packages.filelock: 3.8.0 -> 3.8.2

luizirber committed:
python3Packages.chardet: 5.0.0 -> 5.1.0

luizirber committed:
python3Packages.pyproject-api: init at 1.2.1

luizirber committed:
python3Packages.packaging: 21.3 -> 22.0

luizirber committed:
python3Packages.virtualenv: 20.16.5 -> 20.17.1

luizirber committed:
python3Packages.tox: 3.27.1 -> 4.0.16

luizirber commented:
> Please split into one commit per package with correctly formatted committ messages, e.g. `python310Packages.tox: 3.27.1 -> 4.0.16`.

Broke into one commit per message, and rebased to `python-updates` instead of `staging`.

Should I mark it ready to review and run more CI checks?

mweinelt commented:
I have cherry picked all changes into my local python-updates tree. Building towards tox on staging takes a while and I have to fix issues along the way. Will push the changes later tonight.
